lib/AST/Module.cpp

```diff
@@ -1810,44 +1804,43 @@ void SourceFile::createReferencedNameTracker() {
   ReferencedNames.emplace(ReferencedNameTracker());
 }
 
+ArrayRef<OpaqueTypeDecl *> SourceFile::getOpaqueReturnTypeDecls() {
+  while (!UnvalidatedDeclsWithOpaqueReturnTypes.empty()) {
```

Comment:
I think we can collapse two queues into one by storing the parent decls and kicking off the request on lookup.